worker/modelcache/worker.go

```diff
@@ -324,3 +355,20 @@ func coreNetworkPortRanges(delta []multiwatcher.PortRange) []network.PortRange {
 	}
 	return ports
 }
+
+// lxdCharmProfiler massages a charm.Charm into a LXDProfiler inside of the
+// core package.
+type lxdCharmProfiler struct {
+	Charm charm.Charm
+}
+
+// LXDProfile implements core.lxdprofile.LXDProfiler
+func (p lxdCharmProfiler) LXDProfile() lxdprofile.LXDProfile {
+	if p.Charm == nil {
+		return nil
+	}
+	if profiler, ok := p.Charm.(charm.LXDProfiler); ok {
+		return profiler.LXDProfile()
+	}
+	return nil
+}
```

Comment:
:dagger: we should normalise all the LXDProfiles from charm.v6 to the core/lxdprofile LXDCharmProfiler at some point...